### PR TITLE
fix d_star, #886

### PR DESCRIPTION
**Pull request by 2892510130**
fix D* bug, not it can handle the changed map. #886

### CODE REVIEW

**Comment by 2892510130** [comment]
sorry for the typo, "not" - > "now"

File: PathPlanning/DStar/dstar.py [hunk 1]
```diff
@@ -103,7 +104,7 @@ def process_state(self):
                 if y.h <= k_old and x.h > y.h + x.cost(y):
                     x.parent = y
                     x.h = y.h + x.cost(y)
-        elif k_old == x.h:
+        if k_old == x.h:
             for y in self.map.get_neighbors(x):
```
Comment (AtsushiSakai) [inline PathPlanning/DStar/dstar.py:108]: Why is this change needed?

File: PathPlanning/DStar/dstar.py [hunk 2]
```diff
@@ -116,7 +117,7 @@ def process_state(self):
                     self.insert(y, x.h + x.cost(y))
                 else:
                     if y.parent != x and y.h > x.h + x.cost(y):
-                        self.insert(y, x.h)
+                        self.insert(x, x.h)
```
Comment (AtsushiSakai) [inline PathPlanning/DStar/dstar.py:120]: Was it a bug?

Comment (2892510130) [reply PathPlanning/DStar/dstar.py:120]: Please see https://www.ri.cmu.edu/pub_files/pub3/stentz_anthony__tony__1994_2/stentz_anthony__tony__1994_2.pdf, page 11. Both changes are necessary, otherwise the program will stuck when new obstacles are added. 

Also check out https://www.cs.cmu.edu/~motionplanning/lecture/AppH-astar-dstar_howie.pdf, page 51, I think the instructions on this page indicate the need to change elif to if.

So is this a correct modification？

**Comment by 2892510130** [comment]
Hey there! So what about this PR? I think it is a bug if those two pdfs are correct.

**Comment by AtsushiSakai** [comment]
Sorry. I’m very busy these days, please wait until I have time to review.

**Comment by AtsushiSakai** [comment]
@XwLu @nirnayroy could you guys review this PR?

**Comment by XwLu** [comment]
@AtsushiSakai @2892510130 I have checked the PR with my colleague @David-Randolph, both "elif -> if" and "y -> x" are bugs and necessary to fix.

**Review by AtsushiSakai — APPROVED** [review]
Since one knowledgeable person has approved the changes, I will proceed with the merge.

**Comment by AtsushiSakai** [comment]
@2892510130 Thank you for contribution and sorry for my late response
